coop_memberspace/controllers/main.py

```diff
@@ -0,0 +1,451 @@
+                }
+            )
+        elif user.partner_id.shift_type == "ftop":
+            datas.update({"ftop_point": user.partner_id.display_ftop_points})
+        return request.render("coop_memberspace.mywork", datas)
+
+    @http.route(
+        "/standard/counter_classic", type="http", auth="user", website=True
+    )
+    def page_counter_classic(self, **kwargs):
+        user = request.env.user
+        shift_counter_event_env = request.env["shift.counter.event"]
+        shift_counter_events = shift_counter_event_env.sudo().search(
+            [
+                ("partner_id", "=", user.partner_id.id),
+                ("type", "=", "standard"),
+            ]
+        )
+        return request.render(
+            "coop_memberspace.counter",
+            {
+                "classic": True,
+                "shift_counter_events": shift_counter_events,
+                "user": user,
+            },
+        )
+
+    @http.route(
+        "/standard/counter_extra", type="http", auth="user", website=True
+    )
+    def page_counter_extra(self, **kwargs):
+        user = request.env.user
+        shift_counter_event_env = request.env["shift.counter.event"]
+        shift_counter_events = shift_counter_event_env.sudo().search(
+            [("partner_id", "=", user.partner_id.id), ("type", "=", "ftop")]
+        )
+        return request.render(
+            "coop_memberspace.counter",
+            {
+                "extra": True,
+                "shift_counter_events_extra": shift_counter_events,
+                "user": user,
+            },
+        )
+
+    @http.route(
+        "/standard/programmer_un_extra", type="http", auth="user", website=True
+    )
+    def page_programmer_un_extra(self, **kwargs):
+        user = request.env.user
+        tmpl = user.partner_id.tmpl_reg_line_ids.filtered(
+            lambda r: r.is_current
+        )
+        shift_env = request.env["shift.shift"]
+        shifts_available = shift_env
+        if tmpl:
+            shifts_available = (
+                shift_env.sudo()
+                .search(
+                    [
+                        ("shift_template_id.is_technical", "=", False),
+                        (
+                            "shift_template_id",
+                            "!=",
+                            tmpl[0].shift_template_id.id,
+                        ),
+                        (
+                            "date_begin",
+                            ">=",
+                            (datetime.now() + timedelta(days=1)).strftime(
+                                "%Y-%m-%d 00:00:00"
+                            ),
+                        ),
+                    ]
+                )
+                .filtered(
+                    lambda r, user=request.env.user: user.partner_id
+                    not in r.registration_ids.mapped("partner_id")
+                    and not r.shift_template_id.shift_type_id.is_ftop
+                )
```

Comment:
I'm wondering if we can replace this `filtered` by improving the domain search?

I mean, does it work?  I think it works

```
('registration_ids.partner_id', 'not in', user.partner_id),
('shift_template_id.shift_type_id.is_ftop', '=', False)
```